crates/air-parser/src/parser/lexer/call_variable_parser.rs

```diff
@@ -0,0 +1,242 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use super::LexerError;
+use super::LexerResult;
+use super::Number;
+use super::Token;
+
+use std::iter::Peekable;
+use std::str::CharIndices;
+
+pub(super) fn try_parse_call_variable(
+    string_to_parse: &str,
+    start_pos: usize,
+) -> LexerResult<Token<'_>> {
+    CallVariableParser::new(string_to_parse, start_pos).try_parse()
+}
+
+#[derive(Debug)]
+struct ParserState {
+    pub(self) dot_met_pos: Option<usize>,
+    pub(self) non_numeric_met: bool,
+    pub(self) is_sign_met: bool,
+    pub(self) is_first_char: bool,
+    pub(self) current_char: Option<char>,
+    pub(self) current_pos: Option<usize>,
+}
+
+struct CallVariableParser<'input> {
+    string_to_parse_iter: Peekable<CharIndices<'input>>,
+    string_to_parse: &'input str,
+    start_pos: usize,
+    state: ParserState,
+}
+
+impl<'input> CallVariableParser<'input> {
+    pub(self) fn new(string_to_parse: &'input str, start_pos: usize) -> Self {
+        let string_to_parse_iter = string_to_parse.char_indices().peekable();
+        let state = ParserState {
+            dot_met_pos: None,
+            non_numeric_met: false,
+            is_sign_met: false,
+            is_first_char: true,
+            current_char: None,
+            current_pos: None,
+        };
+
+        Self {
+            string_to_parse_iter,
+            string_to_parse,
+            start_pos,
+            state,
+        }
+    }
+
+    pub(self) fn try_parse(mut self) -> LexerResult<Token<'input>> {
+        while self.next_char() {
+            if self.is_it_possible_to_parse_as_number() {
+                self.try_parse_as_number()?;
+            } else {
+                self.try_parse_as_variable()?;
+            }
+
+            self.state.is_first_char = false;
+        }
+
+        self.to_token()
+    }
+
+    fn next_char(&mut self) -> bool {
+        let (pos, ch) = match self.string_to_parse_iter.next() {
+            Some(pos_and_ch) => pos_and_ch,
+            None => return false,
+        };
+
+        self.state.current_char = Some(ch);
+        self.state.current_pos = Some(pos);
+
+        true
+    }
+
+    fn is_it_possible_to_parse_as_number(&self) -> bool {
+        !self.state.non_numeric_met
+    }
+
+    fn try_parse_as_number(&mut self) -> LexerResult<()> {
+        if self.try_parse_sign() || self.try_parse_digit() || self.try_parse_float_dot() {
+            return Ok(());
+        }
+
+        self.handle_non_digit()
+    }
+
+    fn try_parse_sign(&self) -> bool {
```

Comment:
If a function does unwrap, maybe call it `unchecked` instead of `try`? I propose the following scheme:
1. `try_xx` return LexerResult, unwraps are prohibited
2. `unchecked_xx` or `xx_unchecked` do unwrap